operator/internal/controller/console/controller_test.go

```diff
@@ -0,0 +1,244 @@
+				},
+				Spec: redpandav1alpha2.ConsoleSpec{
+					ClusterSource: &redpandav1alpha2.ClusterSource{
+						StaticConfiguration: &redpandav1alpha2.StaticConfigurationSource{
+							Kafka: &redpandav1alpha2.KafkaAPISpec{
+								Brokers: []string{"kafka-broker:9092"},
+								SASL: &redpandav1alpha2.KafkaSASL{
+									Username:  "testuser",
+									Mechanism: redpandav1alpha2.SASLMechanismPlain,
+									Password: redpandav1alpha2.SecretKeyRef{
+										Name: "kafka-secret",
+										Key:  "password",
+									},
+								},
+							},
+							Admin: &redpandav1alpha2.AdminAPISpec{
+								URLs: []string{"http://admin-api:9644"},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "cluster-ref",
+			console: &redpandav1alpha2.Console{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "console-cluster-ref",
+				},
+				Spec: redpandav1alpha2.ConsoleSpec{
+					ClusterSource: &redpandav1alpha2.ClusterSource{
+						ClusterRef: &redpandav1alpha2.ClusterRef{
+							Name: "test-redpanda",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "no-cluster-source",
+			console: &redpandav1alpha2.Console{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "console-no-cluster",
+				},
+			},
+		},
+		{
+			name: "jwt-set",
+			console: &redpandav1alpha2.Console{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "console-jwt-set",
+				},
+				Spec: redpandav1alpha2.ConsoleSpec{
+					ConsoleValues: redpandav1alpha2.ConsoleValues{
+						Secret: redpandav1alpha2.SecretConfig{
+							Authentication: &redpandav1alpha2.AuthenticationSecrets{
+								JWTSigningKey: ptr.To("some-secret-key"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	ctl := kubetest.NewEnv(t, kube.Options{
+		Options: client.Options{
+			Scheme: controller.UnifiedScheme,
+		},
+	})
+
+	require.NoError(t, kube.ApplyAllAndWait(t.Context(), ctl, func(crd *apiextensionsv1.CustomResourceDefinition, err error) (bool, error) {
+		if err != nil {
+			return false, err
+		}
+
+		for _, cond := range crd.Status.Conditions {
+			if cond.Type == apiextensionsv1.Established {
+				return cond.Status == apiextensionsv1.ConditionTrue, nil
+			}
+		}
+
+		return false, nil
+	}, crds.All()...))
+
+	// Create namespace
+	ns, err := kube.Create(t.Context(), ctl, corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-ns",
+		},
+	})
+	require.NoError(t, err)
+
+	require.NoError(t, ctl.Apply(t.Context(), &redpandav1alpha2.Redpanda{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-redpanda",
+			Namespace: ns.Name,
+		},
+	}))
+
+	consoleCtrl := Controller{
+		Ctl: ctl,
+		rng: rand.New(rand.NewSource(0)),
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create console CR with namespace set
+			console := tc.console.DeepCopy()
+			console.Namespace = ns.Name
+
+			require.NoError(t, ctl.Apply(t.Context(), console))
+
+			// Reconcile the console a few times to ensure determinism.
+			for range 3 {
```

Comment:
@RafalKorepta fixed your previous point by moving everything into the loop.

I've also updated the golden files to include the GVK.